pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Validate the language
+
+			lang := args[1]
+
+			switch lang {
+			case "all":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "nodejs":
+				fmt.Println("Nodejs is supported, publishing")
+			case "python":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "java":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "dotnet":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			default:
+				return fmt.Errorf(
+					"language %s is not supported. Must be one of: \n"+
+						"- all \n"+
+						"- nodejs \n"+
+						"- python \n"+
+						"- java \n"+
+						"- dotnet ", lang,
+				)
+			}
+
+			return nil
+		},
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			pkg := args[0]
+			lang := args[1]
+
+			err = publish(pkg, lang, path, tag)
+			if err != nil {
+				return fmt.Errorf("error publishing: %s", err.Error())
+			}
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	cmd.PersistentFlags().StringVar(&tag, "tag", "",
+		`The tag for your NPM package, such as "latest""`)
+	return cmd
+}
+
+func publish(pkg, lang, path, tag string) error {
+
+	fmt.Printf("You are publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+	case "python":
+		publishToPyPI(pkg)
+	case "java":
+		publishToMaven(pkg)
+	case "dotnet":
+		publishToNuget(pkg)
+	default:
+		// Publish to all registries
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+		publishToPyPI(pkg)
+		publishToMaven(pkg)
+		publishToNuget(pkg)
+	}
+
+	return nil
+}
+
+func publishToNPM(pkg, path, tag string) error {
+	// this is likely where we need to go get the sdk info
+	// verify path
+	_, err := os.Stat(path)
+	if errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("could not find path to package directory: %s", path)
+	}
+
+	// Verify  npm exists set up: npm, user login
+	ex, err := executable.FindExecutable("npm")
+	if err != nil {
+		return fmt.Errorf("could not find npm: %w", err)
+	}
+
+	// TODO: Verify auth for npm
+
+	whoamiCmd := exec.Command(ex, "whoami")
+	whoami, err := whoamiCmd.Output()
+
+	if err != nil {
+		fmt.Print(string(whoami))
+		return fmt.Errorf("make sure you are logged into npm registry")
+	}
+
+	fmt.Printf("Logged in as %s", whoami)
+
+	// TODO: possibly check package dependencies
+
+	var pkgInfo struct {
+		Name    string `json:"name"`
+		Version string `json:"version"`
+	}
+
+	file, err := os.ReadFile(path + "/package.json")
```

Comment:
This should use a filepath.Join so that it uses `\` on Windows.
```suggestion
	file, err := os.ReadFile(filepath.Join(path, "package.json"))
```